api/v1alpha1/halayerset_types.go

```diff
@@ -28,18 +28,38 @@ type HALayerSetSpec struct {
 	// INSERT ADDITIONAL SPEC FIELDS - desired state of cluster
 	// Important: Run "make" to regenerate code after modifying this file
 
-	//Node1Address is the ip address used by the first node in the cluster.
-	Node1Address string `json:"node1address"`
-
-	//Node2Address is the ip address used by the second node in the cluster.
-	Node2Address string `json:"node2address"`
-
-	//NodeName is the name of the node used in current SNO cluster (either cluster1 or cluster2).
-	NodeName string `json:"nodename"`
-
 	//TODO mshitrit use a Secret object
 	//AuthKey is the authkey used by both sno clusters.
 	AuthKey string `json:"authkey"`
+
+	//Deployments is a list of deployments that will be managed by the created HA layer.
+	Deployments []string `json:"deployments,omitempty"`
+
+	//FenceAgentSpec details fence agent specific configuration to use during HA layer fencing setup.
+	FenceAgentSpec FenceAgentSpec `json:"fenceAgentSpec"`
+
+	//NodesSpec contain names and ips of both SNO clusters nodes.
+	NodesSpec NodesSpec `json:"nodesSpec"`
+}
+
+// FenceAgentSpec contains the necessary information for setting up the fence agent that will be used in the HA layer.
+type FenceAgentSpec struct {
+	//Name fence agent name.
+	Name string `json:"name"`
```

Comment:
Probably you also want 
```
Agent string `json:"agent"`
```

Reply:
Since it's a field under `FenceAgentSpec`  I thought that it's obvious we refer to the fence agent,  so `Name` would refer to the fence agent name.
I thought this way it's clearer, pretty much following the same idiom used for [naming conventions](https://golang.org/doc/effective_go#package-names) guide lines that recommended for go packages.